### PR TITLE
FW: rewrite wait_for_child() as wait_for_children()

### PR DESCRIPTION
**Pull request by thiagomacieira**
We only start one child for now, but now we can wait for more than one.

### CODE REVIEW

File: framework/sandstone.cpp [hunk 1]
```diff
@@ -1391,79 +1381,142 @@ static ChildExitStatus wait_for_child(int ffd, intptr_t child, int *tc, const st
...
+                kill(child, SIGKILL);
+        }
+    };
 #elif defined(_WIN32)
-    (void) ffd;
+    auto kill_children = [] { };
+    auto single_wait = [&](milliseconds timeout) {
+        bool bWaitAll = false;
+        auto handles = reinterpret_cast<const HANDLE *>(children.handles.data());
+        DWORD result = WaitForMultipleObjects(children.handles.size(), handles,
```
Comment (thiagomacieira) [inline framework/sandstone.cpp:1430]: [`WaitForMultipleObjects`](https://learn.microsoft.com/en-us/windows/win32/api/synchapi/nf-synchapi-waitformultipleobjects) is limited to `MAXIMUM_WAIT_OBJECTS` objects, which is 64.

That's enough for 2048 cores (16 sockets of 128 or 8 sockets of 256 being the closest, worst-case scenario); after that we'll have a problem.

File: framework/logging.cpp [hunk 2]
```diff
@@ -1645,9 +1645,18 @@ format_duration(MonotonicTimePoint tp, FormatDurationOptions opts = FormatDurati
     return format_duration(tp - sApp->current_test_starttime, opts);
 }
 
-inline AbstractLogger::AbstractLogger(const struct test *test, ChildExitStatus state_)
-    : test(test), childExitStatus(state_), testResult(state_.result)
+static TestResult find_most_serious_result(std::span<const ChildExitStatus> results)
```
Comment (busykai) [inline framework/logging.cpp:1648]: The name sounds funny. Maybe "worst" is better?

**Review by busykai — APPROVED** [review]
This looks good. Cannot tell for the Windows part though.

**Comment by thiagomacieira** [comment]
> This looks good. Cannot tell for the Windows part though.

"It works" suffices for me. We will need to get back to it when we get more than 64 child processes.

**Pull request merged**